### ISSUE TITLE
Newtonsoft 10.0.2 casuing exception

### ISSUE THREAD

**Issue by manniefesto** [sjkp/letsencrypt-siteextension#285]
Hi,

After getting through the initial setup I am getting the following error "Cannot get SerializationBinder because an ISerializationBinder was previously set." on line 153 of the HomeController.

After a little googling it seems that this version of Newtonsoft has known issues when running on Azure. See [https://github.com/Azure/azure-sdk-for-net/issues/3441](url) .

I have manually edited the Web.config and Packages.config to roll back to 9.0.1 and replaced the Newtonsoft DLL's, the extension is now working.

Many thanks,

Edit: Can't get my version numbers right :100:

**Comment by sjkp** [comment]
Which version are you running, and did you do a clean install of it? I can't reproduce this.

**Comment by manniefesto** [comment]
Hi, it was the latest version, I will get more information about the setup tomorrow for you. Thanks.

**Comment by javiercampos** [comment]
Same here, when setting up, on the first "Next" button, I get:
```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Cannot get SerializationBinder because an ISerializationBinder was previously set. at Newtonsoft.Json.JsonSerializer.get_Binder() at Newtonsoft.Json.Serialization.JsonSerializerProxy.get_Binder() --- End of inner exception stack trace --- at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor) at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments) at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, BindingFlags invokeAttr, Binder binder, Object[] index, CultureInfo culture) at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, Object[] index) at Microsoft.Rest.Azure.JsonSerializerExtensions.WithoutConverter(JsonSerializer serializer, JsonConverter converterToExclude) at Microsoft.Rest.Azure.CloudErrorJsonConverter.ReadJson(JsonReader reader, Type objectType, Object existingValue, JsonSerializer serializer) at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.DeserializeConvertable(JsonConverter converter, JsonReader reader, Type objectType, Object existingValue) at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent) at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType) at Microsoft.Rest.Serialization.SafeJsonConvert.DeserializeObject[T](String json, JsonSerializerSettings settings) at Microsoft.Azure.Management.WebSites.WebAppsOperations.<GetWithHttpMessagesAsync>d__7.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.Azure.Management.WebSites.WebAppsOperationsExtensions.<GetAsync>d__5.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.Azure.Management.WebSites.WebAppsOperationsExtensions.Get(IWebAppsOperations operations, String resourceGroupName, String name) at LetsEncrypt.SiteExtension.Controllers.HomeController.Index(AuthenticationModel model) in D:\a\1\s\LetsEncrypt-SiteExtension\Controllers\HomeController.cs:line 40
```

Will try to replace the DLL as the OP said

**Comment by manniefesto** [comment]
Hello all,

Below is the line I have changed in the web config.

`<bindingRedirect oldVersion="0.0.0.0-10.0.0.0" newVersion="9.0.0.0" />`

@sjkp Do you need any further information from me?

Thanks,

**Comment by javiercampos** [comment]
I did the same and it passed the install, but failed later... after installing, I just put it back to 10.0.2 and it seems to work fine

**Comment by dagrooms52** [comment]
I'm hitting the same issue. @manniefesto do you have any resources to share on how you edited Web.config and Packages.config?

**Comment by sjkp** [comment]
I have released a new version 0.9.0 that should solve this,

**Comment by sjkp** [comment]
@dagrooms52 -> what happend to the comment is 0.9.0 still not working for you? (I got an email notification from you)

**Comment by dagrooms52** [comment]
Sorry, just a mistake on my end, your fix in 0.9.0 worked 👍
